### PR TITLE
ENH: support bytearray and unicode (#3)

### PR DESCRIPTION
**Pull request by westurner**
- #3 
- [x] TST: manual test bencode.dumps with Python 2
- [x] TST: manual test bencode.dumps with Python 3
- [x] TST: test that a .torrent file at least loads (with Transmission)
  - [ ] is this handling of utf8 unicode strings consistent with which BEP?
    - https://wiki.theory.org/BitTorrentSpecification#Metainfo_File_Structure
      " All character string values are UTF-8 encoded."
- [x] TST: write test cases for bytearrays
- [ ] TST: write test cases for unicodes
- [ ] PRF: port to _fast.c

### CODE REVIEW

**Comment by notpushkin** [comment]
Fails on Python 3, `NameError: name 'unicode' is not defined` :(

**Comment by westurner** [comment]
Whoops. I think that should only be for Python 2 because in Python 3 all
str are unicode.

**Comment by notpushkin** [comment]
The error actually happens in [the `elif t is unicode`](https://github.com/westurner/better-bencode/blob/62d4d89c79588629546fd24df8d557d85f097d28/better_bencode/_pure.py#L66) line. We can safely substitute unicode for str (in py3k) here (as `unicode_to_bytes` accepts str in python 3), so I suggest this simple fix: https://github.com/Songbee/better-bencode/commit/0bb114b9875fac1a6829e0a7bd8ffb933c66c525

**Comment by westurner** [comment]
Good call @iamale

**Comment by westurner** [comment]
- [ ] TST: What should I add to `TEST_DATA` to test for unicode strings?
- [ ] ENH,TST: IDK Python C-API well enough to port this from _pure.py to _fast.c

**Comment by westurner** [comment]
...  TIL "All character string values are UTF-8 encoded." https://wiki.theory.org/BitTorrentSpecification#Metainfo_File_Structure

File: tests/test_bencode.py [hunk 1]
```diff
@@ -65,6 +65,8 @@ def test_error_dump(module):
     (b'd3:bar4:spam3:fooi42ee', {b'bar': b'spam', b'foo': 42}),
     (b'd1:ai1e1:bi2e1:ci3ee', {b'a': 1, b'b': 2, b'c': 3}),
     (b'd1:a1:be', {b'a': b'b'}),
+    (b'3:\x00\x01\x02', bytearray([0, 1, 2])),
```
Comment (westurner) [inline tests/test_bencode.py:68]: is this correct?

**Comment by notpushkin** [comment]
Also I think it would be nice to decode unicode strings to `unicode`/`str` as well (although it might be a bit trickier).
